tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -917,6 +933,15 @@ private void doCommit(long transactionId, long writePointer, Set<ChangeId> chang
         invalidArray = invalid.toLongArray();
         LOG.info("Tx invalid list: removed committed tx {}", transactionId);
       }
+    } else {
+      long[] checkpointPointers = previous.getCheckpointWritePointers().toLongArray();
```

Comment:
ah yeah. I copied this from existing code and did not attempt to make it better... will do now